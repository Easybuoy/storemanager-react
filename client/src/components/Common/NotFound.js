import React from 'react'
import { Link } from 'react-router-dom';

import NotFoundImage from '../../../assets/images/page-not-found.png';

const NotFound = () => {
  return (
    <div className="text-center">
    <img src={ NotFoundImage } className="notfound"/>
        <div className="text-center">
        <Link to="/dashboard"><button className="button_1 text-center">Back To Dashboard</button></Link>
        </div>
          </div>
  )
}

export default NotFound;