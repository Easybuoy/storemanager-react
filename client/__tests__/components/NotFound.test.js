import React from 'react';
import { shallow } from 'enzyme';

import NotFound from '../../src/components/Common/NotFound';

describe('<NotFound />', () => {
    it('should have 1 img', () => {
        const wrapper = shallow(<NotFound />);
        expect(wrapper.find('img').length).toEqual(1);
    });

    it('renders the NotFound component correctly', () => {
        const wrapper = shallow(<NotFound />);
        expect(wrapper).toMatchSnapshot();
    });
})