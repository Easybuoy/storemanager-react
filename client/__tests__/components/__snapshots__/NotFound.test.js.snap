// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders the NotFound component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="notfound"
          src="test-file-stub"
        />,
        <div
          className="text-center"
        >
          <Link
            replace={false}
            to="/dashboard"
          >
            <button
              className="button_1 text-center"
            >
              Back To Dashboard
            </button>
          </Link>
        </div>,
      ],
      "className": "text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "notfound",
          "src": "test-file-stub",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/dashboard"
          >
            <button
              className="button_1 text-center"
            >
              Back To Dashboard
            </button>
          </Link>,
          "className": "text-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="button_1 text-center"
            >
              Back To Dashboard
            </button>,
            "replace": false,
            "to": "/dashboard",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back To Dashboard",
              "className": "button_1 text-center",
            },
            "ref": null,
            "rendered": "Back To Dashboard",
            "type": "button",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="notfound"
            src="test-file-stub"
          />,
          <div
            className="text-center"
          >
            <Link
              replace={false}
              to="/dashboard"
            >
              <button
                className="button_1 text-center"
              >
                Back To Dashboard
              </button>
            </Link>
          </div>,
        ],
        "className": "text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "notfound",
            "src": "test-file-stub",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/dashboard"
            >
              <button
                className="button_1 text-center"
              >
                Back To Dashboard
              </button>
            </Link>,
            "className": "text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="button_1 text-center"
              >
                Back To Dashboard
              </button>,
              "replace": false,
              "to": "/dashboard",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Back To Dashboard",
                "className": "button_1 text-center",
              },
              "ref": null,
              "rendered": "Back To Dashboard",
              "type": "button",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
